### PR TITLE
feat(es/minifier): remove more functions and classes

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_ecma_minifier/src/compress/optimize/unused.rs [hunk 1]
```diff
@@ -313,7 +313,7 @@ where
 
             Expr::Paren(p) => return self.should_preserve_property_access(&p.expr, opts),
 
-            Expr::Fn(..) | Expr::Arrow(..) | Expr::Array(..) => {
+            Expr::Fn(..) | Expr::Arrow(..) | Expr::Array(..) | Expr::Class(..) => {
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/compress/optimize/unused.rs:316]: Accessing a property of a class can have a side-effect.

Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/unused.rs:316]: Nevermind I found the logic below

File: crates/swc_ecma_minifier/src/compress/optimize/util.rs [hunk 2]
```diff
@@ -122,6 +122,56 @@ pub(crate) fn class_has_side_effect(expr_ctx: &ExprCtx, c: &Class) -> bool {
...
+                ..
+            }) => {
+                if key.expr.may_have_side_effects(expr_ctx) {
+                    res.push(key.expr);
+                }
+            }
+
+            ClassMember::ClassProp(p) => {
+                if let PropName::Computed(key) = p.key {
+                    println!("{:#?}", key.expr.may_have_side_effects(expr_ctx));
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/compress/optimize/util.rs:146]: Please remove this line

File: crates/swc/tests/tsc-references/symbolProperty33_es2015.2.minified.js [hunk 3]
```diff
@@ -1,9 +0,0 @@
-class C1 extends C2 {
```
Comment (kdy1) [inline crates/swc/tests/tsc-references/symbolProperty33_es2015.2.minified.js:1]: `C2` is referenced before the declaration.
We have a relevant field in the analysis result

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

File: crates/swc_ecma_minifier/src/compress/optimize/unused.rs (hunk 1 above)
Comment (Austaras) [reply crates/swc_ecma_minifier/src/compress/optimize/unused.rs:316]: This is actually wrong

File: crates/swc/tests/tsc-references/symbolProperty33_es2015.2.minified.js (hunk 3 above)
Comment (Austaras) [reply crates/swc/tests/tsc-references/symbolProperty33_es2015.2.minified.js:1]: Ah, TDZ. No other passes consider it, and terser ignores it too. Maybe we could fix this later

File: crates/swc_ecma_minifier/tests/postponed.txt [hunk 4]
```diff
@@ -194,6 +194,7 @@ evaluate/unsafe_float_key_complex/input.js
 evaluate/unsafe_integer_key/input.js
 evaluate/unsafe_integer_key_complex/input.js
 evaluate/unsafe_object/input.js
+evaluate/unsafe_object_accessor/input.js
```
Comment (Austaras) [inline crates/swc_ecma_minifier/tests/postponed.txt:197]: Input is syntactically invalid.

File: crates/swc_ecma_minifier/src/compress/optimize/unused.rs (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/unused.rs:316]: Is this code fine?

File: crates/swc_ecma_minifier/tests/full/next-30498/1/output.js [hunk 5]
```diff
@@ -1 +1 @@
-class e extends BaseSchema{matches(e,t){let s=!1,a,r;return t&&("object"==typeof t?{excludeEmptyString:s=!1,message:a,name:r}=t:a=t),this.test({name:r||"matches",message:a||string.matches,params:{regex:e},test:t=>isAbsent(t)||""===t&&s||-1!==t.search(e)})}}(function(){return new e}).prototype=e.prototype
+class e extends BaseSchema{matches(e,t){let s=!1,a,m;return t&&("object"==typeof t?{excludeEmptyString:s=!1,message:a,name:m}=t:a=t),this.test({name:m||"matches",message:a||string.matches,params:{regex:e},test:t=>isAbsent(t)||""===t&&s||-1!==t.search(e)})}}e.prototype
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/full/next-30498/1/output.js:1]: I think we should add `export` to `function string_create()`

File: crates/swc_ecma_minifier/src/compress/optimize/unused.rs (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/unused.rs:316]: I reviewed every other thing including tsc tests, but still worried about this.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

I searched for `should_preserve_property_access` and seems like it's used in various places, and some of them does not have code to handle side effects of a class

File: crates/swc_ecma_minifier/src/compress/optimize/unused.rs (hunk 1 above)
Comment (Austaras) [reply crates/swc_ecma_minifier/src/compress/optimize/unused.rs:316]: Yes, I have patched may_have_side_effect, it's fixed now.

**Review by kdy1 — APPROVED** [review]
I see, thank you!

---

swc-bump:
 - swc_ecma_utils
 - swc_ecma_minifier

**Review by kdy1 — APPROVED** [review]

**Pull request merged**